### ISSUE TITLE
Should not succeed for empty CMakeLists.txt

### ISSUE THREAD

**Issue by mloskot** [richq/cmake-lint#33]
Version: `cmakelint 1.3.4.1`

It should not report `0` errors for empty thus invalid `CMakeLists.txt`:

```
$ cat CMakeLists.txt

$ cmakelint CMakeLists.txt
Total Errors: 0
```

**Comment by tkruse** [comment]
In what way is an empty CMakeLists.txt invalid? cmake 3.10.2 process it without error. (It doesn't do much, but that's not the same as defining it as invalid).

```
$ cmake --version
cmake version 3.10.2
$ cat CMakeLists.txt 

$ cmake CMakeLists.txt 
-- Configuring done
-- Generating done
-- Build files have been written to:
```

**Comment by tkruse** [comment]
Actually even if it was invalid, cmakelint does not necessarily have to report anything. 
cmakelints purpose is to report style issues. Now whether an all-empty cmake file is a style issue or not I am not sure.

Maybe if you explain your situation more it would be easier to understand what solution would be best.

**Comment by mloskot** [comment]
https://cmake.org/cmake/help/latest/policy/CMP0000.html#policy:CMP0000

> CMake requires that projects specify the version of CMake to which they have been written. (...) Note that the command invocation must appear in the CMakeLists.txt file itself; (...)

I'm surprised `cmake` can swallow empty `CMakeLists.txt`. I'd consider it a bug. Since `cmake` does not reject empty script, then nothing to blame `cmakelint` for.
